test/samplers/test_samplers.py

```diff
@@ -150,59 +165,87 @@ def test_random_sampler_randomness():
     assert builtin_random_state_start == builtin_random_state_end
 
 
-def test_random_sampler_errors():
+@pytest.mark.parametrize(
+    "num_clips, sampling_range_size",
+    (
+        # Ask for 50 clips while the sampling range is 10 frames wide
+        # expect 10 clips with 10 unique starting points.
+        (10, 10),
+        # Ask for 50 clips while the sampling range is only 10 frames wide
+        # expect 50 clips with only 10 unique starting points.
+        (50, 10),
+    ),
+)
+def test_sample_at_regular_indices_num_clips_large(num_clips, sampling_range_size):
+    # Test for expected behavior described in Note [num clips larger than sampling range]
+    decoder = VideoDecoder(NASA_VIDEO.path)
+    clips = clips_at_regular_indices(
+        decoder,
+        num_clips=num_clips,
+        sampling_range_start=0,
+        sampling_range_end=sampling_range_size,  # because sampling_range_start=0
+    )
+
+    assert len(clips) == num_clips
+
+    clip_starts_seconds = torch.tensor([clip.pts_seconds[0] for clip in clips])
+    assert len(torch.unique(clip_starts_seconds)) == sampling_range_size
+
+    # Assert clips starts are ordered, i.e. the start indices don't just "wrap
+    # around". They're duplicated *and* ordered.
```

Comment:
Mention reason for not wrapping around?

Reply:
There's no particular reason honestly, it's just an assertion of the behavior of `torch.linspace`. I'll add a comment near the Note to clarify that this is somewhat arbitrary.